docs/sources/tempo/configuration/network/tls.md

```diff
@@ -96,3 +96,127 @@ The above structure can be set on the following receiver configurations:
 - `distributor.receivers.zipkin.tls`
 - `distributor.receivers.jaeger.protocols.grpc.tls`
 - `distributor.receivers.jaeger.protocols.thrift_http.tls`
+
+### Configure TLS with Helm
+
+To configure TLS with the Helm chart, you must have a TLS key-pair and CA certificate stored in a Kubernetes secret. The following example mounts a secret called `tempo-distributed-tls` into the pods at `/tls` and modifies the configuration of Tempo to make use of the files. In this example the Tempo components share a single TLS certificate. Note that the `tls_server_name` configuration must match the certificate.
+
+```yaml
+compactor:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+distributor:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+ingester:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+metricsGenerator:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+querier:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+queryFrontend:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+tempo:
+  readinessProbe:
+    httpGet:
+      scheme: HTTPS
+  structuredConfig:
```

Comment:
this section should also have, or a mention of it in the written part

```
cache:
  caches:
    - memcached:
      <tls_config>
```

Reply:
This works and I'll include a bit of config for it in the example.  The exporter is a little more work due to the way the annotations are handled, so I'm going to leave that out of this PR.